mathesar_ui/src/sections/table-view/actions-pane/RenameTableModal.svelte

```diff
@@ -0,0 +1,97 @@
+<script lang='ts'>
+  import { tick } from 'svelte';
+  import { CancelOrProceedButtonPair, Modal, TextInput } from '@mathesar-component-library';
+  import type { TabularData } from '@mathesar/stores/table-data/types';
+  import { refetchTablesForSchema, renameTable, tables } from '@mathesar/stores/tables';
+  import { toast } from '@mathesar/stores/toast';
+  import { currentSchemaId } from '@mathesar/stores/schemas';
+  import { constructTabularTab, getTabsForSchema } from '@mathesar/stores/tabs';
+  import { currentDBName } from '@mathesar/stores/databases';
+  import { TabularType } from '@mathesar/App.d';
+
+  export let isOpen: boolean;
+  export let tabularData: TabularData;
+
+  let allowClose = true;
+  let inputElement: HTMLInputElement;
+  let originalName = '';
+  let name = '';
+  let proceed: () => Promise<void>;
+
+  function schemaContainsTableName(_name: string): boolean {
+    return [...$tables.data.values()].map((t) => t.name).includes(_name);
```

Comment:
This is the offending line that prompted me to set `'@typescript-eslint/no-unsafe-return': 'off',`

I'm a little surprised that eslint struggled here. But in general I want to move in the direction of turning off linting rules when they're getting in our way.

Here it seems like `svelte-check` should catch problems. For example, I tried changing the return statement to `return 7` and observed the following error from `svelte-check`:

> Error: Type 'number' is not assignable to type 'boolean'. (ts)

To me, that result from `svelte-check` says that we don't need eslint for this particular rule.

Reply:
I'm not sure what could have caused eslint to fail here, I assume it has something to do with recognizing the type of `$tables`.

I'm okay with disabling problematic rules for now.